lib/ssh/setup-ssh-agent

```diff
@@ -0,0 +1,9 @@
+sshdir=$(getent passwd $USER | cut -d: -f6)/.ssh
+mkdir $sshdir
+chmod 700 $sshdir
+cat > $sshdir/key
+chmod 600 $sshdir/key
+eval "$(ssh-agent -s)"
+gha-set-env SSH_AUTH_SOCK <<< $SSH_AUTH_SOCK
+gha-set-env SSH_AGENT_PID <<< $SSH_AGENT_PID
+ssh-add $sshdir/key
```

Comment:
If you're going to give an argument to `ssh-add` (I honestly forgot you could do that) then you don't actually need to mess with sshdir at all, or even write the private key to disk, which actually seems preferable.


```
ssh-add -  # activate private key on stdin
```

Reply:
switched to reading the key from stdin.